sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_common/_base_prompty_eval.py

```diff
@@ -70,7 +71,15 @@ async def _do_eval(self, eval_input: Dict) -> Dict[str, float]:
 
         score = math.nan
         if llm_output:
+            if self._result_key in PROMPT_BASED_REASON_EVALUATORS:
```

Comment:
Nit: This code block is odd enough without context that I think it merits a small explanation.
```suggestion
            # Parse out score and reason from evaluators known to possess them.
            if self._result_key in PROMPT_BASED_REASON_EVALUATORS:
```